src/app/features/task-repeat-cfg/store/task-repeat-cfg.effects.ts

```diff
@@ -58,6 +66,131 @@
     { dispatch: false },
   );
 
+  /**
+   * Updates the repeatCfg of a task, if the task was updated.
+   */
+  updateRepeatCfgWhenTaskUpdates: any = createEffect(
+    () =>
+      this._actions$.pipe(
+        ofType(updateTask),
+        tap(async (aAction) => {
+          const allTasks = await this._taskService.allTasks$.pipe(first()).toPromise();
+          const task = allTasks.find((aTask) => aTask.id === aAction.task.id)!;
+
+          if (task.repeatCfgId !== null) {
+            const repeatCfgForTask = await this._taskRepeatCfgService
+              .getTaskRepeatCfgById$(task.repeatCfgId)
+              .pipe(first())
+              .toPromise();
+
+            const taskChanges = aAction.task.changes;
+
+            // TODO: is there a better way to do this? Is there anything missing?
+            const repeatCfgChanges: Partial<TaskRepeatCfgCopy> = {
+              projectId: taskChanges.projectId ?? repeatCfgForTask.projectId,
+              title: taskChanges.title ?? repeatCfgForTask.title,
+              tagIds: taskChanges.tagIds ?? repeatCfgForTask.tagIds,
+              notes: taskChanges.notes ?? repeatCfgForTask.notes,
+            };
+
+            // TODO: Do we need to do this for all instances??
+            this._taskRepeatCfgService.updateTaskRepeatCfg(
+              task.repeatCfgId,
+              repeatCfgChanges,
+            );
+          }
+        }),
+      ),
+    { dispatch: false }, // Question: What exactly does this do?
```

Comment:
The 'dispatch: false' in the createEffect function means that the effect will not dispatch any new actions. If you want to dispatch new actions from within the effect, you should remove this option or set it to true. [medium]